spec/classes/candlepin_qpid_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'spec_helper'
+
+describe 'candlepin::config' do
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) { facts }
+
+      describe 'without parameters' do
+        let :pre_condition do
+          "class {'candlepin':
+            qpid_ssl_cert => '/path/to/ssl/cert'
+          }"
+        end
+
+        describe "should setup qpid event exchange" do
+          it {
+            should contain_qpid__config__exchange('event').
+                    with('exchange' => 'event').
```

Comment:
I've been using the new style which I like a bit more:
```ruby
is_expected.to contain_qpid__config__exchange('event')
  .with_exchange('event')
  .with_hostname('localhost')
```

It has the benefit that when you add a line you're not changing the one above it. I didn't know you could specify the with like you do with the arrow. Yours feels less magic.

You're also missing a check on the ssl_key.